testing/test_inference.py

```diff
@@ -0,0 +1,176 @@
+                             collate_fn=mt_datasets.mt_collate,
+                             num_workers=1)
+
+    if film_bool:
+        model = torch.load(os.path.join(PATH_BIDS, "model_film_test.pt"), map_location=device)
+    else:
+        model = torch.load(os.path.join(PATH_BIDS, "model_unet_test.pt"), map_location=device)
+
+    if cuda_available:
+        model.cuda()
+    model.eval()
+
+    metric_fns = [dice_score,  # from ivadomed/utils.py
+                  mt_metrics.hausdorff_score,
+                  mt_metrics.precision_score,
+                  mt_metrics.recall_score,
+                  mt_metrics.specificity_score,
+                  mt_metrics.intersection_over_union,
+                  mt_metrics.accuracy_score]
+
+    metric_mgr = IvadoMetricManager(metric_fns)
+
+    if not os.path.isdir(PATH_OUT):
+        os.makedirs(PATH_OUT)
+
+    pred_tmp_lst, z_tmp_lst, fname_tmp = [], [], ''
+    for i, batch in enumerate(test_loader):
+        input_samples, gt_samples = batch["input"], batch["gt"]
+
+        with torch.no_grad():
+            if cuda_available:
+                test_input = input_samples.cuda()
+                test_gt = gt_samples.cuda(non_blocking=True)
+            else:
+                test_input = input_samples
+                test_gt = gt_samples
+
+            if film_bool:
+                sample_metadata = batch["input_metadata"]
+                test_contrast = [sample_metadata[k]['contrast'] for k in range(len(sample_metadata))]
+
+                test_metadata = [one_hot_encoder.transform([sample_metadata[k]["film_input"]]).tolist()[0] for k in
+                                 range(len(sample_metadata))]
+                preds = model(test_input, test_metadata)  # Input the metadata related to the input samples
+            else:
+                preds = model(test_input)
+
+        # WARNING: sample['gt'] is actually the pred in the return sample
+        # implementation justification: the other option: rdict['pred'] = preds would require to largely modify mt_transforms
+        rdict = {}
+        rdict['gt'] = preds.cpu()
+        batch.update(rdict)
+
+        # reconstruct 3D image
+        for smp_idx in range(len(batch['gt'])):
+            # undo transformations
+            rdict = {}
+            for k in batch.keys():
+                rdict[k] = batch[k][smp_idx]
+            rdict_undo = val_undo_transform(rdict)
+
+            fname_ref = rdict_undo['input_metadata']['gt_filename']
+            if pred_tmp_lst and (fname_ref != fname_tmp or (i == len(test_loader)-1 and smp_idx == len(batch['gt'])-1)):  # new processed file
+                # save the completely processed file as a nii
+                fname_pred = os.path.join(PATH_OUT, fname_tmp.split('/')[-1])
+                fname_pred = fname_pred.split('manual.nii.gz')[0] + 'pred.nii.gz'
+                save_nii(pred_tmp_lst, z_tmp_lst, fname_tmp, fname_pred, SLICE_AXIS)
```

Comment:
@charleygros I've changed `model_unet_test.pt` and fixed the problem with Cuda in `test_inference.py`, but I got a new error with [save_nii(pred_tmp_lst, z_tmp_lst, fname_tmp, fname_pred, SLICE_AXIS)](https://github.com/neuropoly/ivado-medical-imaging/blob/cg/reconstruct-3D/testing/test_inference.py#L150).
 
--> here: [pytest](https://github.com/neuropoly/ivado-medical-imaging/runs/346353620#step:6:156).